File: sdks/typescript/examples/streaming-chat.ts
import { UarClient } from "../src/index.js";
const client = new UarClient(process.env.UAR_URL ?? "http://localhost:1906");
for await (const event of client.chat.stream({ messages: [{ role: "user", content: "Stream a haiku" }] })) console.log(event);

File: sdks/rust/examples/streaming_chat.rs
use futures::StreamExt;
use universal_agent_runtime_sdk::{ChatCompletionRequest, ChatMessage, Client};
#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    let client = Client::new(
        std::env::var("UAR_BASE_URL").unwrap_or_else(|_| "http://localhost:1906".into()),
    )?;
    let mut events = client
        .chat()
        .stream(ChatCompletionRequest {
            messages: vec![ChatMessage::text("user", "Tell me a short story")],
            ..Default::default()
        })
        .await?;
    while let Some(event) = events.next().await {
        println!("{:#?}", event?);
    }
    Ok(())
}
